airflow-core/src/airflow/exceptions.py

```diff
@@ -511,3 +511,35 @@ def __str__(self):
 
 class UnknownExecutorException(ValueError):
     """Raised when an attempt is made to load an executor which is not configured."""
+
+
+class AirflowFileLockAcquireException(AirflowException):
+    """Raised when a file lock cannot be acquired (e.g., via `fcntl.flock` with LOCK_SH or LOCK_EX)."""
+
+    def __init__(self, file_path: str | None, original_exception: Exception | None = None):
+        super().__init__()
+        self.file_path = file_path
+        self.original_exception = original_exception
```

Comment:
We don't need to handle the original exception for us -- the `raise ... from e` means the original exception will already be recorded in `self.__cause__` (and displayed for us by pthon)